src/corehost/cli/args.cpp

```diff
@@ -53,6 +53,16 @@ void setup_shared_store_paths(const hostpolicy_init_t& init, const pal::string_t
         args->dotnet_shared_store = own_dir;
         append_path(&args->dotnet_shared_store, RUNTIME_STORE_DIRECTORY_NAME);
         append_path(&args->dotnet_shared_store, init.tfm.c_str());
+
+        //Also add an architecture specific path to allow for unzipped developer cli experience
+        //where developer points runtime store operation output to "dotnet.exe" relative shared store folder
+
+        pal::string_t local_developer_store = own_dir ;
+        append_path(&local_developer_store, RUNTIME_STORE_DIRECTORY_NAME);
+        append_path(&local_developer_store, get_arch());
+        append_path(&local_developer_store, init.tfm.c_str());
+        local_developer_store.push_back(PATH_SEPARATOR);
+        (args->env_shared_store).push_back(local_developer_store);
```

Comment:
The ordering being described above by @eerhardt also does not lineup with the multi-level lookup we have for SharedFX and SDK (see https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/multilevel-sharedfx-lookup.md). 

Why do we need the entry @eerhardt calls out above to be added. Why cannot it be the non-architecture entry (penultimate in the list) and everything just works? Arch is only required for userprofile case.